### PR TITLE
Vulkan: Don't do MoltenVK things on Asahi Linux

### PR DESCRIPTION
**Pull request by TellowKrinkle**
Honeykrisp (Mesa Vulkan driver for Apple GPUs) is in fact *not* MoltenVK, and doesn't appreciate being given things that are only accepted by our custom modified MoltenVK

~~Note: Haven't actually tested this on Apple + MoltenVK yet, will test soon~~ Tested, fbfetch seems to be working fine

### CODE REVIEW

**Review by K0bin — APPROVED** [review]
Looks fine overall.

I'm not a fan of the `VULKAN_CONTEXT_ENUMERATE_WANTED_FEATURES` macro though. That saves time typing it once but is just more difficult to read after that.

**Comment by TellowKrinkle** [comment]
> I'm not a fan of the VULKAN_CONTEXT_ENUMERATE_WANTED_FEATURES macro though. That saves time typing it once but is just more difficult to read after that.

Removed

File: Source/Core/VideoBackends/Vulkan/VulkanContext.cpp [hunk 1]
```diff
@@ -22,22 +22,101 @@ static constexpr const char* VALIDATION_LAYER_NAME = "VK_LAYER_KHRONOS_validatio
...
+    : m_instance(instance), m_physical_device(physical_device), m_device_info(physical_device)
 {
-  // Read device physical memory properties, we need it for allocating buffers
-  vkGetPhysicalDeviceProperties(physical_device, &m_device_properties);
-  vkGetPhysicalDeviceMemoryProperties(physical_device, &m_device_memory_properties);
-
-  // Would any drivers be this silly? I hope not...
-  m_device_properties.limits.minUniformBufferOffsetAlignment = std::max(
-      m_device_properties.limits.minUniformBufferOffsetAlignment, static_cast<VkDeviceSize>(1));
-  m_device_properties.limits.minTexelBufferOffsetAlignment = std::max(
```
Comment (iwubcode) [inline Source/Core/VideoBackends/Vulkan/VulkanContext.cpp:35]: Some of these did not make their way into `PhysicalDeviceInfo` was that intentional?

Comment (TellowKrinkle) [reply Source/Core/VideoBackends/Vulkan/VulkanContext.cpp:35]: Yes, I only put things we actually use into PhysicalDeviceInfo
If those end up needed in the future, we can add them then
If you think those should have been used... well I guess we should use them then

**Review by iwubcode — APPROVED** [review]

**Comment by mackal** [comment]
Getting crashes with this (vulkan backend only) on Linux. Tried the commit before this and no issues (issues on head as well, but tried this specific commit) others in the support-linux discord are also having issues.

**Comment by iwubcode** [comment]
Of course, this happens when I decide not to test 😆 .  This impacts Windows as well, I am hitting this:  https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/blob/master/include/vk_mem_alloc.h#L12986 .

**Comment by iwubcode** [comment]
We explicitly set [VMA_VULKAN_VERSION](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/VideoBackends/Vulkan/VulkanLoader.h#L66).  But seems like it will be auto detected based on [vulkan include version](https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/blob/master/include/vk_mem_alloc.h#L135), so not sure why we did that.

**Pull request merged**